src/DynamicRupture/FrictionLaws/BaseFrictionLaw.h

```diff
@@ -45,105 +39,266 @@ class seissol::dr::friction_law::BaseFrictionLaw {
    */
   void postcomputeImposedStateFromNewStress(
       real QInterpolatedPlus[CONVERGENCE_ORDER][tensor::QInterpolated::size()],
       real QInterpolatedMinus[CONVERGENCE_ORDER][tensor::QInterpolated::size()],
       const FaultStresses& faultStresses,
       double timeWeights[CONVERGENCE_ORDER],
-      unsigned int ltsFace);
+      unsigned int ltsFace) {
+    // this initialization of the kernel could be moved to the initializer
+    // set inputParam could be extendent for this (or create own function)
+    // the kernel then could be a class attribute and following values are only set once
+    //(but be careful of race conditions since this is computed in parallel for each face!!)
+    dynamicRupture::kernel::ImposedStateFromNewStress ImposedStateFromNewStressKrnl;
+    ImposedStateFromNewStressKrnl.select0 = init::select0::Values;
+    ImposedStateFromNewStressKrnl.select3 = init::select3::Values;
+    ImposedStateFromNewStressKrnl.select5 = init::select5::Values;
+    ImposedStateFromNewStressKrnl.select6 = init::select6::Values;
+    ImposedStateFromNewStressKrnl.select7 = init::select7::Values;
+    ImposedStateFromNewStressKrnl.select8 = init::select8::Values;
+    ImposedStateFromNewStressKrnl.inv_Zs = impAndEta[ltsFace].inv_Zs;
+    ImposedStateFromNewStressKrnl.inv_Zs_neig = impAndEta[ltsFace].inv_Zs_neig;
+    ImposedStateFromNewStressKrnl.inv_Zp = impAndEta[ltsFace].inv_Zp;
+    ImposedStateFromNewStressKrnl.inv_Zp_neig = impAndEta[ltsFace].inv_Zp_neig;
+
+    // set imposed state to zero
+    for (unsigned int i = 0; i < tensor::QInterpolated::size(); i++) {
+      imposedStatePlus[ltsFace][i] = 0;
+      imposedStateMinus[ltsFace][i] = 0;
+    }
+    ImposedStateFromNewStressKrnl.imposedStatePlus = imposedStatePlus[ltsFace];
+    ImposedStateFromNewStressKrnl.imposedStateMinus = imposedStateMinus[ltsFace];
+
+    for (int j = 0; j < CONVERGENCE_ORDER; j++) {
+      ImposedStateFromNewStressKrnl.NorStressGP = faultStresses.NormalStressGP[j];
+      ImposedStateFromNewStressKrnl.TractionGP_XY = faultStresses.XYTractionResultGP[j];
+      ImposedStateFromNewStressKrnl.TractionGP_XZ = faultStresses.XZTractionResultGP[j];
+      ImposedStateFromNewStressKrnl.timeWeights = timeWeights[j];
+      ImposedStateFromNewStressKrnl.QInterpolatedMinus = QInterpolatedMinus[j];
+      ImposedStateFromNewStressKrnl.QInterpolatedPlus = QInterpolatedPlus[j];
+      // Carsten Uphoff Thesis: EQ.: 4.60
+      ImposedStateFromNewStressKrnl.execute();
+    }
+  }
 
-  /*
-   * https://strike.scec.org/cvws/download/SCEC_validation_slip_law.pdf
+  /**
+   * For reference, see: https://strike.scec.org/cvws/download/SCEC_validation_slip_law.pdf
    */
-  real calcSmoothStepIncrement(real current_time, real dt);
+  real calcSmoothStepIncrement(real currentTime, real dt) {
+    if (currentTime > 0.0 && currentTime <= drParameters.t0) {
+      real gNuc = calcSmoothStep(currentTime);
+      real prevTime = currentTime - dt;
+      if (prevTime > 0.0) {
```

Comment:
Not exactly: If `t0 < currentTime < t0+dt`, then it does acutally make a difference. But still it would be more correct, if we just leave out the ifs...

Reply:
maybe we should apply the changes also on master to avoid small differences when comparing both versions?.